src/EnergyPlus/DataZoneEquipment.cc

```diff
@@ -1724,4 +1733,256 @@ void scaleInletFlows(EnergyPlusData &state, int const zoneNodeNum, int const spa
+                                             Real64 const nonAirSysOutput,
+                                             int const equipTypeNum)
+{
+    for (auto &splitterSpace : this->spaces) {
+        if (this->tstatControl != DataZoneEquipment::ZoneEquipTstatControl::Ideal) {
+            // Restore zone loads to unadjusted values (for all ZoneEquipTstatControl types except Ideal)
+            state.dataZoneEnergyDemand->ZoneSysEnergyDemand(zoneNum) = this->saveZoneSysSensibleDemand;
+            state.dataZoneEnergyDemand->ZoneSysMoistureDemand(zoneNum) = this->saveZoneSysMoistureDemand;
+        }
+
+        Real64 spaceFraction = splitterSpace.fraction;
+        if (this->tstatControl == DataZoneEquipment::ZoneEquipTstatControl::Ideal) {
+            // Proportion output by sensible space load / zone load (varies every timestep, overrides outputFraction)
+            auto &thisZoneSysEnergyDemand = state.dataZoneEnergyDemand->ZoneSysEnergyDemand(zoneNum);
+            if (thisZoneSysEnergyDemand.RemainingOutputRequired != 0.0) {
+                spaceFraction = state.dataZoneEnergyDemand->spaceSysEnergyDemand(splitterSpace.spaceIndex).RemainingOutputRequired /
+                                thisZoneSysEnergyDemand.RemainingOutputRequired;
+            }
+        }
+
+        Real64 spaceSysOutputProvided = sysOutputProvided * spaceFraction;
+        Real64 spaceLatOutputProvided = latOutputProvided * spaceFraction;
+        state.dataZoneTempPredictorCorrector->spaceHeatBalance(splitterSpace.spaceIndex).NonAirSystemResponse += nonAirSysOutput * spaceFraction;
+        if (this->zoneEquipOutletNodeNum > 0 && splitterSpace.spaceNodeNum > 0) {
+            auto &equipOutletNode = state.dataLoopNodes->Node(this->zoneEquipOutletNodeNum);
+            auto &spaceInletNode = state.dataLoopNodes->Node(splitterSpace.spaceNodeNum);
+            spaceInletNode.MassFlowRate = equipOutletNode.MassFlowRate * spaceFraction;
+            spaceInletNode.MassFlowRateMaxAvail = equipOutletNode.MassFlowRateMaxAvail * spaceFraction;
+            spaceInletNode.MassFlowRateMinAvail = equipOutletNode.MassFlowRateMinAvail * spaceFraction;
+            spaceInletNode.Temp = equipOutletNode.Temp;
+            spaceInletNode.HumRat = equipOutletNode.HumRat;
+            spaceInletNode.CO2 = equipOutletNode.CO2;
+        }
+        ZoneEquipmentManager::updateSystemOutputRequired(state,
+                                                         zoneNum,
+                                                         spaceSysOutputProvided,
+                                                         spaceLatOutputProvided,
+                                                         state.dataZoneEnergyDemand->spaceSysEnergyDemand(splitterSpace.spaceIndex),
+                                                         state.dataZoneEnergyDemand->spaceSysMoistureDemand(splitterSpace.spaceIndex),
+                                                         equipTypeNum);
+    }
+}
```

Comment:
These are some nice new functions, clean with switch blocks and references.  Good stuff!